src/neural/cuda/network_cudnn.cc

```diff
@@ -198,6 +198,11 @@ class CudnnNetwork : public Network {
                    pblczero::NetworkFormat::POLICY_CONVOLUTION;
 
     max_batch_size_ = options.GetOrDefault<int>("max_batch", 1024);
+    // Batches smaller than 4 have different results for same input for
+    // cudnn-fp16 mode. Deterministic flag allows this to be disabled by forcing
+    // all batches to pretend to be at least 4 in size.
+    min_batch_size_ =
+        options.GetOrDefault<bool>("deterministic", false) ? 4 : 1;
```

Comment:
maybe make it `true` by default?

Reply:
I assume there is performance cost to pay there - not sure if its significant though.

Reply:
I think it's pretty rare that backend ends up evaluating a tiny batch, so I guess it should be fine.. probably.